integration-tests/smoke/ccip_messaging_test.go

```diff
@@ -165,18 +171,159 @@ func Test_CCIPMessaging(t *testing.T) {
+		out.msgSentEvent.Message.Header.MessageId,
+	)
+	tree, err := merklemulti.NewTree(hashutil.NewKeccak(), [][32]byte{messageHash})
+	require.NoError(t, err)
+	proof, err := tree.Prove([]int{0})
+	require.NoError(t, err)
+	require.Equal(t, merkleRoot, tree.Root())
+
+	tx, err := state.Chains[destChain].OffRamp.ManuallyExecute(
+		e.Env.Chains[destChain].DeployerKey,
+		[]offramp.InternalExecutionReport{
+			{
+				SourceChainSelector: sourceChain,
+				Messages: []offramp.InternalAny2EVMRampMessage{
+					{
+						Header: offramp.InternalRampMessageHeader{
+							MessageId:           out.msgSentEvent.Message.Header.MessageId,
+							SourceChainSelector: sourceChain,
+							DestChainSelector:   destChain,
+							SequenceNumber:      out.msgSentEvent.SequenceNumber,
+							Nonce:               out.msgSentEvent.Message.Header.Nonce,
+						},
+						Sender:       sender,
+						Data:         []byte("hello CCIPReceiver with low exec gas"),
+						Receiver:     state.Chains[destChain].Receiver.Address(),
+						GasLimit:     big.NewInt(1),
+						TokenAmounts: []offramp.InternalAny2EVMTokenTransfer{},
+					},
+				},
+				OffchainTokenData: [][][]byte{
+					{},
+				},
+				Proofs:        proof.Hashes,
+				ProofFlagBits: boolsToBitFlags(proof.SourceFlags),
+			},
+		},
+		[][]offramp.OffRampGasLimitOverride{
+			{
+				{
+					ReceiverExecutionGasLimit: big.NewInt(200_000),
+					TokenGasOverrides:         nil,
+				},
+			},
+		},
+	)
+	_, err = deployment.ConfirmIfNoError(e.Env.Chains[destChain], tx, err)
+	require.NoError(t, err, "failed to send/confirm manuallyExecute tx")
+
+	receipt, err := e.Env.Chains[destChain].Client.TransactionReceipt(ctx, tx.Hash())
+	require.NoError(t, err)
+
+	var found bool
+	for _, lg := range receipt.Logs {
+		topic := offramp.OffRampExecutionStateChanged{}.Topic()
+		if lg.Topics[0] == topic {
+
+			parsed, err := state.Chains[destChain].OffRamp.ParseExecutionStateChanged(*lg)
+			require.NoError(t, err)
+
+			require.Equal(t, sourceChain, parsed.SourceChainSelector)
+			require.Equal(t, out.msgSentEvent.SequenceNumber, parsed.SequenceNumber)
+			require.Equal(t, out.msgSentEvent.Message.Header.MessageId, parsed.MessageId)
+			require.Equal(t, messageHash, parsed.MessageHash)
+			require.Equal(t, uint8(ccdeploy.EXECUTION_STATE_SUCCESS), parsed.State)
+			found = true
+		}
+	}
+	require.True(t, found, "no ExecutionStateChanged event found in manual execution receipt logs")
+}
+
+func getMerkleRoot(
+	t *testing.T,
+	offRamp *offramp.OffRamp,
+	seqNr uint64,
+) (merkleRoot [32]byte) {
+	iter, err := offRamp.FilterCommitReportAccepted(nil)
+	require.NoError(t, err)
+	for iter.Next() {
+		for _, mr := range iter.Event.MerkleRoots {
+			if mr.MinSeqNr >= seqNr || mr.MaxSeqNr <= seqNr {
+				merkleRoot = mr.MerkleRoot
+				break
+			}
+		}
+	}
+	require.NotEqualf(
+		t,
+		[32]byte{},
+		merkleRoot,
+		"no merkle root found for sequence number %d",
+		seqNr,
+	)
+	return merkleRoot
+}
```

Comment:
nit: if you return early the not found check is a little simpler
```suggestion
func getMerkleRoot(
	t *testing.T,
	offRamp *offramp.OffRamp,
	seqNr uint64,
) (merkleRoot [32]byte) {
	iter, err := offRamp.FilterCommitReportAccepted(nil)
	require.NoError(t, err)
	for iter.Next() {
		for _, mr := range iter.Event.MerkleRoots {
			if mr.MinSeqNr >= seqNr || mr.MaxSeqNr <= seqNr {
				return mr.MerkleRoot
			}
		}
	}
	require.Fail(
		t,
		"no merkle root found for sequence number %d",
		seqNr,
	)
	return merkleRoot
}
```

Reply:
Absolutely right, applied